### PR TITLE
Update ListingsView.js

### PR DESCRIPTION
**Pull request by JorgeVidal29**
This is a refact of listing view to remove the need for a forceUpdate

**Description**
Wanted to remove the need for a forceUpdate

Resolves #(replace parentheses and the stuff here with the issue number, if there isn't one, create one.) 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/marketplace/ListingsView.js [hunk 1]
```diff
@@ -54,7 +58,7 @@ class ListingsView extends React.Component<Props, State> {
 
 	//Gets all the sells and filters them based on what type we are currently looking at
 	getMarketSales = async ():Promise<void> => {
-		itemsForSale = [];
+		let itemsForSaleArray = [];
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: Make it const please. You can still push to it (the thing that is constant is that you don't set it to a new array; the array itself can change)

File: src/marketplace/ListingsView.js [hunk 2]
```diff
@@ -10,15 +10,19 @@ type Props = {|
 |};
 type State = {|
 	userId: string,
+	//This allows for all the items that are for sale to be with in one array
+	itemsForSale: Array<Object>
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: Thanks for putting it in the state, but it isn't just an array of any type of object. You are making some assumptions about what will be in this array, so those should be indicated to other people in the Array<whateverTheTypeShouldBe> part

**Review by SecondThread — CHANGES_REQUESTED** [review]

File: src/marketplace/ListingsView.js (hunk 1 above)
Comment (JorgeVidal29) [reply src/marketplace/ListingsView.js]: Done

File: src/marketplace/ListingsView.js (hunk 2 above)
Comment (JorgeVidal29) [reply src/marketplace/ListingsView.js]: Did it with a interface

File: src/marketplace/ListingsView.js [hunk 3]
```diff
@@ -76,27 +89,25 @@ class ListingsView extends React.Component<Props, State> {
 				obj['amount'] = jsonObjectOfSells[sale].amount;
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: So why not make saleObject a class? Then you don't have to be messing with object[string] and you can just call the constructor and not worry about all of this?

File: src/marketplace/ListingsView.js [hunk 4]
```diff
@@ -4,21 +4,34 @@ import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
 import Cookies from 'universal-cookie';
 import type {ComponentType} from '../armory/ComponentType.js';
 
+
+interface itemForSaleObject {
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: err, correct me if I'm wrong, but this seems like exactly the job for a class, no? Usually you would use an interface if you have a bunch of classes that all share a couple methods but are different. All of these are just items for sale, right?

**Comment by Derplime** [comment]
Fixed the merge conflict with `master`. All you need to do is address the comments David made.

File: src/marketplace/ListingsView.js [hunk 5]
```diff
@@ -69,34 +74,26 @@ class ListingsView extends React.Component<Props, State> {
...
 			{
-				//This is a temp obj used to create an object of what a sell item would need to have. It is pushed into an array of all of the items for sell that fall in the current seller type
-				let obj = {};
-				obj['name'] = jsonObjectOfSells[sale].itemId;
-				obj['price'] = jsonObjectOfSells[sale].salePrice;
-				obj['amount'] = jsonObjectOfSells[sale].amount;
-				obj['sellerId'] = jsonObjectOfSells[sale].sellerId;
-				obj['saleId'] = jsonObjectOfSells[sale]._id;
-				itemsForSale.push(obj);
+				const sellingObject = new saleObject(jsonObjectOfSells[sale].itemId,jsonObjectOfSells[sale].salePrice,jsonObjectOfSells[sale].amount,jsonObjectOfSells[sale].sellerId,jsonObjectOfSells[sale]._id);
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: This line is exceedingly long; please separate into multiple lines

File: src/marketplace/ListingsView.js [hunk 6]
```diff
@@ -3,22 +3,27 @@ import * as React from 'react';
 import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
 import Cookies from 'universal-cookie';
 import type {ComponentType} from '../armory/ComponentType.js';
+import saleObject from './saleObjectClass.js';
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: Class names should be UpperCamelCase

File: src/marketplace/ListingsView.js [hunk 7]
```diff
@@ -3,22 +3,27 @@ import * as React from 'react';
...
-//This allows for all the items that are for sale to be with in one array
-let itemsForSale = [{name: '', price: 0, amount: 0, sellerId: '', saleId: ''}];
+
 class ListingsView extends React.Component<Props, State> {
 	constructor(props:Props) {
 		super(props);
 		this.state={
 			userId: '',
+			//Each item sale object will have a name, a price, an amount, a seller id, and the sale id
+			//itemsForSale : [{name: '', price: 0, amount: 0, sellerId: '', saleId: ''}]
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: delete these two lines if they aren't relevant anymore please

File: src/marketplace/ListingsView.js [hunk 8]
```diff
@@ -69,34 +74,26 @@ class ListingsView extends React.Component<Props, State> {
 			const typeOfItem = getTankComponent(verifyComponent(jsonObjectOfSells[sale].itemId));
 			if(typeOfItem === this.props.sellerType)
 			{
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: Fix to match style guide

File: src/marketplace/ListingsView.js [hunk 9]
```diff
@@ -69,34 +74,26 @@ class ListingsView extends React.Component<Props, State> {
...
 				<div className="card mb-2" key={i}>
 					<div className="card-body">
-						<h5 className="card-title">Item to buy: {itemsForSale[i].name}</h5>
-						<h5 className="card-title">Price: ${itemsForSale[i].price}</h5>
-						<h5 className="card-title">Quantity: {itemsForSale[i].amount}</h5>
-						<button className="btn btn-success mt-2" onClick={this.buyItem.bind(this,itemsForSale[i].sellerId, itemsForSale[i].saleId)}>Buy</button>
+						<h5 className="card-title">Item to buy: {this.state.itemsForSale[i].name}</h5>
+						<h5 className="card-title">Price: ${this.state.itemsForSale[i].price}</h5>
+						<h5 className="card-title">Quantity: {this.state.itemsForSale[i].amount}</h5>
+						<button className="btn btn-success mt-2" onClick={this.buyItem.bind(this,this.state.itemsForSale[i].sellerId, this.state.itemsForSale[i].saleId)}>Buy</button>
```
Comment (SecondThread) [inline src/marketplace/ListingsView.js]: You can make this line cleaner and easier to read with ES6:
`onClick={() => this.buyItem(this.state.itemsForSale[i].sellerId, this.state.itemsForSale[i].saleId)}`

File: src/marketplace/saleObjectClass.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+//@flow strict
+class saleObject {
+	name:string;
```
Comment (SecondThread) [inline src/marketplace/saleObjectClass.js:4]: fix spacing please

File: src/marketplace/saleObjectClass.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+//@flow strict
+class saleObject {
```
Comment (SecondThread) [inline src/marketplace/saleObjectClass.js]: mentioned earlier, but UpperCamelCase please :)

File: src/marketplace/ListingsView.js (hunk 8 above)
Comment (SecondThread) [reply src/marketplace/ListingsView.js]: also fix line 72/73

**Review by SecondThread — COMMENTED** [review]
No significant improvements, just several small things

**Comment by SecondThread** [comment]
* no significant changes requested. I didn't mean that there weren't significant improvements to the PR; there certainly were

**Review by SecondThread — APPROVED** [review]
Thanks for fixing. As long as it passes flow and is tested, the code looks good to me :)

**Pull request merged**